Develop

haduigon opened the pull request:
I modified the data to create an opportunity to change the color of the product. Additionally, instead of creating separate pages for different products, I created one pattern for all. All features seem to be realized.

SanyaBratashchuk reviewed (changes requested):
please provide demo link

haduigon commented:
Hi, everyone again. The situation is next : 
I have modified the input data to have opportunity changing the product color inside productCard. I put modified data inside api folder and it did nit work when deployed on git hub pages due to localhost as far as I nave understood. So, I added one test route to my server, it always gives the same data.json which is in the api folder under customProducts.json. Also instead of writing different element for products I tried to create one for all. Thank you in advance.

anastasiiavorobiova reviewed (changes requested):
Great work! <img width="305" alt="Screenshot 2024-03-17 at 15 46 14" src="https://github.com/mate-academy/react_phone-catalog/assets/135232001/2ba5d858-dcb1-4f9e-87df-ea514feba03d">
The image should be clickable on the card 
<img width="1470" alt="Screenshot 2024-03-17 at 15 43 47" src="https://github.com/mate-academy/react_phone-catalog/assets/135232001/76a17ffd-be1d-4826-a9c9-c1f1a3028b8f">
The search doesn't work. No need to show it on the product page 
<img width="268" alt="Screenshot 2024-03-17 at 15 42 31" src="https://github.com/mate-academy/react_phone-catalog/assets/135232001/fbe3dd62-ea10-4fc9-bc18-d0bd235566ff">
The link should lead to the home page. If you feel some comments are unclear, please ask for help in the chat or come to the Q&A session. We will deal with everything)

haduigon commented:
> Great work! <img alt="Screenshot 2024-03-17 at 15 46 14" width="305" src="https://private-user-images.githubusercontent.com/135232001/313477306-2ba5d858-dcb1-4f9e-87df-ea514feba03d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA2ODU1NjcsIm5iZiI6MTcxMDY4NTI2NywicGF0aCI6Ii8xMzUyMzIwMDEvMzEzNDc3MzA2LTJiYTVkODU4LWRjYjEtNGY5ZS04N2RmLWVhNTE0ZmViYTAzZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxN1QxNDIxMDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jNzQxYjMzYzdjMGUxY2E3ZWNmMmNjYmE5MmYwNWVjOTdlMzMyZjA4MmUyZDJjMjI3NTFlNjNhYzY2ZjI5ZDg0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.sjhLj6TThtCpH4MsO56P2LvYY71iPEWSR2ZgTfhsVUk"> The image should be clickable on the card <img alt="Screenshot 2024-03-17 at 15 43 47" width="1470" src="https://private-user-images.githubusercontent.com/135232001/313477348-76a17ffd-be1d-4826-a9c9-c1f1a3028b8f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA2ODU1NjcsIm5iZiI6MTcxMDY4NTI2NywicGF0aCI6Ii8xMzUyMzIwMDEvMzEzNDc3MzQ4LTc2YTE3ZmZkLWJlMWQtNDgyNi1hOWM5LWMxZjFhMzAyOGI4Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxN1QxNDIxMDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zYWJiZTJlNzMxYzY3MDg5MjNiMjc5NDFjMTk3OTY0ZTk3MjJhZTRjMzFmY2Q0YWM1ZWUwYzNkMzJiZDk5NmQwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.7N3uavbXGqtMh2N9mwzoAeYHQvJ091CmfWTDTadVnZY"> The search doesn't work. No need to show it on the product page <img alt="Screenshot 2024-03-17 at 15 42 31" width="268" src="https://private-user-images.githubusercontent.com/135232001/313477413-fbe3dd62-ea10-4fc9-bc18-d0bd235566ff.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA2ODU1NjcsIm5iZiI6MTcxMDY4NTI2NywicGF0aCI6Ii8xMzUyMzIwMDEvMzEzNDc3NDEzLWZiZTNkZDYyLWVhMTAtNGZjOS1iYzE4LWQwYmQyMzU1NjZmZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxN1QxNDIxMDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05MjcwNzYwMjc4YzE5NWRjMzlhYTBmNmViNzBmNjA3MjNiNDNjOTQ4ZDk2NDEyZWUwOGUyMmFhMDQ4OGU5OTU1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Gl00evIDr4w1HT6VJ2rnyfmsPi73hEGYxvSpViNR9pM"> The link should lead to the home page. If you feel some comments are unclear, please ask for help in the chat or come to the Q&A session. We will deal with everything)

everything is done )

haduigon commented:
do you see my today`s commits?

Esceype reviewed (changes requested):
GJ!
Need to fix:
- add a favicon;
- "Shop by category" section redirect to bottom of the page in some section, but it should redirect to the top. Also in  "Shop by category" images should redirect too.
- Search still not work it doesn`t filter phones but instead somewhy redirect us back to home page;
- Here should be cursor pointer on hover:
![image](https://github.com/mate-academy/react_phone-catalog/assets/64101898/60ca164d-0222-42c4-b5a1-5e48f23911fb)
 and same for cross button;
- Checkout button doesn`t work. I think this button should clear our card and somehow notify the user that they have successfully placed an order;
- Logo should be clickable and should redirect user to home page;
- If "Favorites" is empty then there should be some text about it like it is in cart page;

haduigon commented:
> GJ! Need to fix:
> 
> * add a favicon;

I don`t have a favicon, so I added a red heart icon

> * "Shop by category" section redirect to bottom of the page in some section, but it should redirect to the top. Also in  "Shop by category" images should redirect too.

fixed
> * Search still not work it doesn`t filter phones but instead somewhy redirect us back to home page;

finally fixed, it seems
> * Here should be cursor pointer on hover:
>   ![image](https://private-user-images.githubusercontent.com/64101898/314145967-60ca164d-0222-42c4-b5a1-5e48f23911fb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE4MjA4NjIsIm5iZiI6MTcxMTgyMDU2MiwicGF0aCI6Ii82NDEwMTg5OC8zMTQxNDU5NjctNjBjYTE2NGQtMDIyMi00MmM0LWI1YTEtNWU0OGYyMzkxMWZiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzMwVDE3NDI0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgyYzM4NWVlZjVmOGM4MDljYTlhOWQ2NmZiMmJmZGVjY2NlZTFlZTM5ZTkwYWFhYWI2MDNiMjg0ZmNmYmYyZDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.PWtOd-kzxQmVNOzSmPPVf8qOmAM8CJpg4ApSOcrpmlc)

done
>   and same for cross button;

done
> * Checkout button doesn`t work. I think this button should clear our card and somehow notify the user that they have successfully placed an order;

Made it as it was described in the task :
Checkout button should show a modal dialog with the text Checkout is not implemented yet. Do you want to clear the Cart?:
Clear the Cart if the user confirms the order;
Keep the Cart items and close the confirmation on cancel;

> * Logo should be clickable and should redirect user to home page;

done

> * If "Favorites" is empty then there should be some text about it like it is in cart page;
++


Also I noticed that there were added mobile design, I tried ti replicate.

haduigon commented:
після деплою на гиті помитів, що не відтворюються зображення, та чомусь, не підтягнувся фавікон, на локалхості все норм
<img width="1654" alt="Screenshot at Apr 01 15-00-54" src="https://github.com/mate-academy/react_phone-catalog/assets/20277989/2079d2d5-ea26-4fc3-9ee8-197d364775f9">

IvanFesenko reviewed (changes requested):
- favIcon is still missing, somesthing wrong with banner
![2024-04-01_15h16_15](https://github.com/mate-academy/react_phone-catalog/assets/61904995/4d1cc2a7-89fd-418b-942b-2c6b76717f48)
- Look weird that header and footer has different sizes
![2024-04-01_15h18_27](https://github.com/mate-academy/react_phone-catalog/assets/61904995/4a1b2cf8-216c-4b65-805d-c860ce690774)
![2024-04-01_15h18_46](https://github.com/mate-academy/react_phone-catalog/assets/61904995/c2c7a163-8f85-4415-8f76-9abe3d863725)
- Should be links - not divs
![2024-04-01_15h20_08](https://github.com/mate-academy/react_phone-catalog/assets/61904995/e0fef5e2-2626-460f-a460-c66768a1baeb)
- Slider is broken
![2024-04-01_15h21_14](https://github.com/mate-academy/react_phone-catalog/assets/61904995/6c21fe19-bbbd-4806-9462-8498a8041c72)
- Check this case
![2024-04-01_15h22_18](https://github.com/mate-academy/react_phone-catalog/assets/61904995/1ed2bdb4-b69a-4329-a69f-44ab4f2e7dab)
- Pagination issue 
https://github.com/mate-academy/react_phone-catalog/assets/61904995/804311e3-ed93-437c-874d-9e3661792c95
- Search not working on Favorites page, save favorites in localStorage as well
![2024-04-01_15h25_27](https://github.com/mate-academy/react_phone-catalog/assets/61904995/21db158f-873c-46cc-96c4-975c8713c33c)
- What is this text for?
![2024-04-01_15h26_26](https://github.com/mate-academy/react_phone-catalog/assets/61904995/314ab205-0120-4006-a205-c0e7ea9b1382)
- Switcher not working
![2024-04-01_15h27_04](https://github.com/mate-academy/react_phone-catalog/assets/61904995/68cdfa50-7d72-44a8-99f5-6a61b8422d42)
- Why tablets page shows phones?
![2024-04-01_15h27_46](https://github.com/mate-academy/react_phone-catalog/assets/61904995/feb217ff-fdfe-42de-9ab8-3a7282a5b990)
- Make text more consistent
![2024-04-01_15h28_23](https://github.com/mate-academy/react_phone-catalog/assets/61904995/05130050-19f0-4967-b068-b6847babe911)
- Looks like same models, check your displaying logic
![2024-04-01_15h30_56](https://github.com/mate-academy/react_phone-catalog/assets/61904995/a7a1450e-7cd0-4fdb-b832-ab8649a1b907)

haduigon commented:
https://github.com/mate-academy/react_phone-catalog/assets/20277989/2f2d5f70-1da4-4a9f-9a9e-549b01959b54

не можу відтворити помилку з рядами

haduigon commented:
> * favIcon is still missing, somesthing wrong with banner

fixed

>   ![2024-04-01_15h16_15](https://private-user-images.githubusercontent.com/61904995/318427971-4d1cc2a7-89fd-418b-942b-2c6b76717f48.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxNTkzNzgsIm5iZiI6MTcxMjE1OTA3OCwicGF0aCI6Ii82MTkwNDk5NS8zMTg0Mjc5NzEtNGQxY2MyYTctODlmZC00MThiLTk0MmItMmM2Yjc2NzE3ZjQ4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAzVDE1NDQzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQyOTNkODUxY2Q2NGYyM2M5ODI4YmRkYmQ2NjUwZTlhYTIxNThjNTBhYzc3MzA2MTk2MmRkNjRiOTNkYmE4MDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.KxbpptANclSZ3RF1P0v11kGA8qnyyiSLfQYIJ3hAqC4)
> * Look weird that header and footer has different sizes
done


>   ![2024-04-01_15h18_27](https://private-user-images.githubusercontent.com/61904995/318428409-4a1b2cf8-216c-4b65-805d-c860ce690774.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxNTkzNzgsIm5iZiI6MTcxMjE1OTA3OCwicGF0aCI6Ii82MTkwNDk5NS8zMTg0Mjg0MDktNGExYjJjZjgtMjE2Yy00YjY1LTgwNWQtYzg2MGNlNjkwNzc0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAzVDE1NDQzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ2Yjc2ZThjYzYzZGI2MDUxZDI4YzE2ZDdhZTc5NjkxYzU1ZTgzZjRjNDc4NTJiYjJlMmZkMDY1ZGU1YzVlNTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.xjFDvYE8S53JYa6hpme7iP0_KlCnisQ8iodA7HyIrzw)
>   ![2024-04-01_15h18_46](https://private-user-images.githubusercontent.com/61904995/318428415-c2c7a163-8f85-4415-8f76-9abe3d863725.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxNTkzNzgsIm5iZiI6MTcxMjE1OTA3OCwicGF0aCI6Ii82MTkwNDk5NS8zMTg0Mjg0MTUtYzJjN2ExNjMtOGY4NS00NDE1LThmNzYtOWFiZTNkODYzNzI1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAzVDE1NDQzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg3MjU5NTFmZTJiNzUyNjU0MjAwMjcwY2NhZGU3YjUzOWJmMjE3NjNiNWFmNzYzNDMxZWNkMTM2ZjljN2VjNjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6Wm4ZqqDUPgTsoaJfsE1Zp05M9M9mdXmutNMrS7EtdI)
> * Should be links - not divs

done

>   ![2024-04-01_15h20_08](https://private-user-images.githubusercontent.com/61904995/318428641-e0fef5e2-2626-460f-a460-c66768a1baeb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxNTkzNzgsIm5iZiI6MTcxMjE1OTA3OCwicGF0aCI6Ii82MTkwNDk5NS8zMTg0Mjg2NDEtZTBmZWY1ZTItMjYyNi00NjBmLWE0NjAtYzY2NzY4YTFiYWViLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAzVDE1NDQzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFkMTY3ZDE5NDMzMWIxNDk2NDUyZjhhYzBjYzgyZmVmYTFjYmQxMTcyZTcwZmU5ZjAxODM5Mjk2ZDhlZjZhODQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NEbfshLCoL1dkOBssQQWXkW7zqivs5BFNlhWmhJX7mk)
> * Slider is broken

done

>   ![2024-04-01_15h21_14](https://private-user-images.githubusercontent.com/61904995/318428802-6c21fe19-bbbd-4806-9462-8498a8041c72.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxNTkzNzgsIm5iZiI6MTcxMjE1OTA3OCwicGF0aCI6Ii82MTkwNDk5NS8zMTg0Mjg4MDItNmMyMWZlMTktYmJiZC00ODA2LTk0NjItODQ5OGE4MDQxYzcyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAzVDE1NDQzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU2N2QyZGU1MDdiNTMxNmYyYjhiNGQ5ZDFmMjBiMDdhMjFiYmRiYzUxNjdhMjE2ZmQ3MTlmN2MzOTA3ZmRkY2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.YnPnoANtqx--EpUicy3i_jCqtGbKc5BDnyMY17gEa2I)
> * Check this case

done

>   ![2024-04-01_15h22_18](https://private-user-images.githubusercontent.com/61904995/318428991-1ed2bdb4-b69a-4329-a69f-44ab4f2e7dab.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxNTkzNzgsIm5iZiI6MTcxMjE1OTA3OCwicGF0aCI6Ii82MTkwNDk5NS8zMTg0Mjg5OTEtMWVkMmJkYjQtYjY5YS00MzI5LWE2OWYtNDRhYjRmMmU3ZGFiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAzVDE1NDQzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI5ZGYxNjBiNjFlMGE3OTJlN2Q1YzlkNzc2ZjljOThlMWY5YzZlYzgxMmUwYTI5MGQ3NDRhZDFhOTFkNDZhNzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.9lvV_emwlULe_CRu2qmm9HbWOnnd37LnTMf_xg1VRzc)
> * Pagination issue

done

>   https://github.com/mate-academy/react_phone-catalog/assets/61904995/804311e3-ed93-437c-874d-9e3661792c95
> * Search not working on Favorites page,

done

 save favorites in localStorage as well

there is no functional logic how to delete favourites if they are saved to localStorage it could be eternal preferences )))


>   ![2024-04-01_15h25_27](https://private-user-images.githubusercontent.com/61904995/318429604-21db158f-873c-46cc-96c4-975c8713c33c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxNTkzNzgsIm5iZiI6MTcxMjE1OTA3OCwicGF0aCI6Ii82MTkwNDk5NS8zMTg0Mjk2MDQtMjFkYjE1OGYtODczYy00NmNjLTk2YzQtOTc1Yzg3MTNjMzNjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAzVDE1NDQzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFiNzBkN2RmNjI3YTk4ODcwYTRhYWY3NWU4MzgxMTQwMmI1MDFhMmUwODc3OWQwM2FiZTljM2I3NzgxYWM5YzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.t5F7fAzVVsxlMwDwIx1XvL0zU6ms2iyYr-EBThv0COg)
> * What is this text for?

it is a phone name  as in the design example



>   ![2024-04-01_15h26_26](https://private-user-images.githubusercontent.com/61904995/318429722-314ab205-0120-4006-a205-c0e7ea9b1382.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxNTkzNzgsIm5iZiI6MTcxMjE1OTA3OCwicGF0aCI6Ii82MTkwNDk5NS8zMTg0Mjk3MjItMzE0YWIyMDUtMDEyMC00MDA2LWEyMDUtYzBlN2VhOWIxMzgyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAzVDE1NDQzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJjNWMxYjkyYzAyOWFjMTFkOTE1NjZlZjBiNWI3ZWNjYTg5OGU3OTViNGI3MzQ2MzllMWUzYTYwZmI3M2Y4NWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-E0N7tvhZ2kZ2FkZ76zqWWti4OOOdRp6Kkiitp7ZRSo)
> * Switcher not working

the thing is that it should not working:

Show the details on the page:
Fix one Available color and one Select capacity value for now;

I realized as I have understood it there is no description how exactly it should wrk


>   ![2024-04-01_15h27_04](https://private-user-images.githubusercontent.com/61904995/318429877-68cdfa50-7d72-44a8-99f5-6a61b8422d42.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxNTkzNzgsIm5iZiI6MTcxMjE1OTA3OCwicGF0aCI6Ii82MTkwNDk5NS8zMTg0Mjk4NzctNjhjZGZhNTAtN2Q3Mi00NGE4LTk5ZjUtNmE2MWI4NDIyZDQyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAzVDE1NDQzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY2OTE0Mjg4OWVjYmExMTdlZDE3NGVkMGUwNjUxNDNhYjk4OGIzNGNiZjI3MzllYzIxYTRiMjVlNTJhYjA3ZDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Zx1OZgy-21_VajqYRLK6SV0C3OY9ukT9tWBsM4B4PmA)
> * Why tablets page shows phones?

it shows tablets, the only one difference is the type which is shown here :  

<img width="1079" alt="tabl" src="https://github.com/mate-academy/react_phone-catalog/assets/20277989/9314079f-cf5b-4176-aeab-803fcd3b44e7">
<img width="1048" alt="phons" src="https://github.com/mate-academy/react_phone-catalog/assets/20277989/b661335b-70b2-4bd1-b1ad-2de2658393ac">


>   ![2024-04-01_15h27_46](https://private-user-images.githubusercontent.com/61904995/318429979-feb217ff-fdfe-42de-9ab8-3a7282a5b990.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxNTkzNzgsIm5iZiI6MTcxMjE1OTA3OCwicGF0aCI6Ii82MTkwNDk5NS8zMTg0Mjk5NzktZmViMjE3ZmYtZmRmZS00MmRlLTlhYjgtM2E3MjgyYTViOTkwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAzVDE1NDQzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVjZjc3YWVkM2NjNjJiNDdkNTY2MGY5N2ZmZGM0NzYyOGZmOTlmNmIwY2ZjMWU5ZDg4ZmFmMGZhZTQ2OTZhYzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ZwtS9Ufd12WVPmGIfnzfp5H-eOs56TWLTIDgKxTFXfQ)
> * Make text more consistent

>   ![2024-04-01_15h28_23](https://private-user-images.githubusercontent.com/61904995/318430093-05130050-19f0-4967-b068-b6847babe911.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxNTkzNzgsIm5iZiI6MTcxMjE1OTA3OCwicGF0aCI6Ii82MTkwNDk5NS8zMTg0MzAwOTMtMDUxMzAwNTAtMTlmMC00OTY3LWIwNjgtYjY4NDdiYWJlOTExLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAzVDE1NDQzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThiMGQ0NTVhMmMwMGYzNjQ4YTJiZTIxZjYxMzA3YWMyMjZlYTRkZTJjOWUzMTYxMDUxMjVkODU2Yzg1Y2NkNTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.14AfriIPVqZXkVXQzFyLJGCe4dNipgaA17LdBG9I9q0)
> * Looks like same models, check your displaying logic

they are not :  

<img width="1372" alt="1" src="https://github.com/mate-academy/react_phone-catalog/assets/20277989/aea3727e-b060-4e26-8dc4-5c23406ece0e">


>   ![2024-04-01_15h30_56](https://private-user-images.githubusercontent.com/61904995/318431077-a7a1450e-7cd0-4fdb-b832-ab8649a1b907.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxNTkzNzgsIm5iZiI6MTcxMjE1OTA3OCwicGF0aCI6Ii82MTkwNDk5NS8zMTg0MzEwNzctYTdhMTQ1MGUtN2NkMC00ZmRiLWI4MzItYWI4NjQ5YTFiOTA3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAzVDE1NDQzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZjZTcxMjVlYWRiY2Q4OTE4MDE3NDU5MTI0NzJmMDY0NWU0MmRkMWVlMDJhZmZlNjlhMTQ4OTBmMDE4NWE4MjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.guVgDyv6Tp1Krhyqd-oHfQB6qzuQC6dasL2B_fYhXMM)
